lib/datadog/tracing/distributed/trace_context.rb

```diff
@@ -142,6 +147,11 @@ def build_trace_flags(digest)
         # @see https://www.w3.org/TR/trace-context/#tracestate-header
         def build_tracestate(digest)
           tracestate = String.new('dd=')
+          unless digest.is_remote
```

Comment:
Following up from this PR https://github.com/DataDog/dd-trace-rb/pull/3516, it looks like `digest.is_remote` will almost always be `false`; it will only be `true` if we are injecting a recently extracted `TraceDigest`. All other `TraceDigest`s will have `digest.is_remote == false`, because they were created internally.

Reply:
Yup that's the desired behavior. We expect the first case to be hit the vast majority of the time:
- If a TraceDigest is created from an internal span then TraceDigest.span_id is the last datadog span id in the trace
- If a TraceDigest represents a span from a remote host then the TraceDigest.span_id may not be the last datadog span id in the trace. In this scenario we must check the incoming for the last datadog span id (`p:.......` field).
- If the digest represents a remote span and the last datadog parent id is not propagated in the incoming tracestate then do nothing. We can not set the last datadog span id, this information has been lost.

Reply:
I just added tests for this logic